### DISCUSSION TITLE
Vanna is not able to get data from local Postgres DB

### DISCUSSION THREAD

**Discussion by imranrazakhan** [vanna-ai/vanna#576] [Q&A]
I have modified streamlit app to use ollama and chromadb
 
 **vanna_calls.py**
 
```
from vanna.ollama import Ollama
from vanna.chromadb import ChromaDB_VectorStore

class MyVanna(ChromaDB_VectorStore, Ollama):
    def __init__(self, config=None):
        ChromaDB_VectorStore.__init__(self, config=config)
        Ollama.__init__(self, config=config)

@st.cache_resource(ttl=3600)
def setup_vanna():
    vn = MyVanna(config={'model': 'llama3:8b-instruct-q8_0', 'ollama_host': 'http://192.x.x.x:11434'})

    vn.connect_to_postgres(host='x.x.x.x', dbname='energy_db', user='postgres', password='xxx', port='5432')
    # The information schema query may need some tweaking depending on your database. This is a good starting point.
    df_information_schema = vn.run_sql("SELECT * FROM INFORMATION_SCHEMA.COLUMNS")
    plan = vn.get_training_plan_generic(df_information_schema)
    vn.train(plan=plan)
    return vn
```

It runs fine without error, but dont pick data from database, 

When i ask question in logs i can see

`SQL Prompt: [{'role': 'system', 'content': "You are a PostgreSQL expert. Please help to generate a SQL query to answer the question. Your response should ONLY be based on the given context and follow the response guidelines and format instructions. \n===Additional Context \n\nThe following columns are in the pg_tables table in the energy_db database`

but results always come from model without trained data

<img width="656" alt="image" src="https://github.com/user-attachments/assets/19dc3c83-aaf5-4290-8cab-39386ebd79e6">

**Desktop (please complete the following information where):**
 - OS: [e.g. Ubuntu]
 - Version: [e.g. 20.04]
 - Python: [3.12]
 - Vanna: [0.6.4]

**Comment by zainhoda** [comment]
You have to use a chat variant. You can't use `llama3:8b-instruct-q8_0`

**Reply by imranrazakhan** [reply]
@zainhoda Thanks for the reply; it was really helpful. I tried CodeLlama, Mistral, and others, but ALIENTELLIGENCE/pythoncoder outperformed them all in my scenarios. The responses were much quicker compared to the others.
